File: TRNMNT/ClientApp/app/core/model/category.model.ts
import {WeightDivisionModel} from "./weight-division.model"

export class CategoryModel {

    constructor() {
        this.weightDivisions = []
    }

    public name: string
    public categoryId?: AAGUID;
    public eventId: string;
    public weightDivisions: WeightDivisionModel[];
}

File: TRNMNT.Core/Model/CategoryModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TRNMNT.Core.Model
{
    public class CategoryModel
    {
        public Guid CategoryId { get; set; }
        public String Name { get; set; }
        public Guid EventId { get; set; }

        public ICollection<WeightDivisionModel> WeightDivisions { get; set; }
    }
}
